spec/presenters/candidate_interface/application_form_presenter_spec.rb

```diff
@@ -32,6 +32,22 @@
     end
   end
 
+  describe '#contact_details_valid?' do
+    it 'returns true if contact details section is completed' do
+      application_form = FactoryBot.build(:application_form, contact_details_completed: true)
+      presenter = CandidateInterface::ApplicationFormPresenter.new(application_form)
+
+      expect(presenter).to be_contact_details_valid
+    end
+
+    it 'returns true if contact details section is incomplete' do
+      application_form = FactoryBot.build(:application_form, contact_details_completed: false)
+      presenter = CandidateInterface::ApplicationFormPresenter.new(application_form)
+
+      expect(presenter).to be_contact_details_valid
+    end
```

Comment:
```suggestion
    end

    it 'returns false if contact details section is invalid' do
      application_form = FactoryBot.build(:application_form, phone_number: '')
      presenter = CandidateInterface::ApplicationFormPresenter.new(application_form)

      expect(presenter).to be_contact_details_valid
    end
```

Reply:
Hey would this not be :

`expect(presenter).to_not be_contact_details_valid`

I've tried to implement this and for some reason it keeps coming through as valid... which was the issue with the precious solutions I was trying.

Reply:
🤦 my bad!